lib/ash_json_api/serializer.ex

```diff
@@ -179,7 +179,14 @@ defmodule AshJsonApi.Serializer do
   end
 
   defp many_links(request, %{results: _} = paginator) do
-    uri = URI.parse(request.url)
+    path =
+      if request.json_api_prefix do
```

Comment:
This is surprising. `request.url` should be right here. What's the incorrect value you are seeing for `request.url`?

Reply:
I logged the value of request.url and Its the bare url.
```elixir
    IO.inspect(request.url)
    # => "http://localhost:4000"
    uri = URI.parse(request.url)
```
Called from
```sh
curl 'localhost:4000/api/lol/participants/list_replayable' | jq "."
...
  "links": {
    "first": "http://localhost:4000?page[limit]=10",
    "self": "http://localhost:4000?page[limit]=10",
    "next": "http://localhost:4000?page[after]=g2wAAAACdAAAAA13C21pY3Jvc2Vjb25kaAJiAAeB4WEGdwZzZWNvbmRhGXcIY2FsZW5kYXJ3E0VsaXhpci5DYWxlbmRhci5JU093BW1vbnRoYQN3Cl9fc3RydWN0X193D0VsaXhpci5EYXRlVGltZXcDZGF5YQp3BHllYXJiAAAH6HcGbWludXRlYS93BGhvdXJhCHcJdGltZV96b25lbQAAAAdFdGMvVVRDdwl6b25lX2FiYnJtAAAAA1VUQ3cKdXRjX29mZnNldGEAdwpzdGRfb2Zmc2V0YQBtAAAAJGFkZjllMzQ0LTgyMjktNDU5YS1iZmFlLTdmN2I5Nzc4NWM0NWo%3D&page[limit]=10",
    "prev": null
  },
...
```
Patching `ash_json_api` and recompiling with the fix
```sh
  "links": {
    "first": "http://localhost:4000/api/lol/participants/list_replayable?page[limit]=10",
    "self": "http://localhost:4000/api/lol/participants/list_replayable?page[limit]=10",
    "next": "http://localhost:4000/api/lol/participants/list_replayable?page[after]=g2wAAAACdAAAAA13C21pY3Jvc2Vjb25kaAJiAAeB4WEGdwZzZWNvbmRhGXcIY2FsZW5kYXJ3E0VsaXhpci5DYWxlbmRhci5JU093BW1vbnRoYQN3Cl9fc3RydWN0X193D0VsaXhpci5EYXRlVGltZXcDZGF5YQp3BHllYXJiAAAH6HcGbWludXRlYS93BGhvdXJhCHcJdGltZV96b25lbQAAAAdFdGMvVVRDdwl6b25lX2FiYnJtAAAAA1VUQ3cKdXRjX29mZnNldGEAdwpzdGRfb2Zmc2V0YQBtAAAAJGFkZjllMzQ0LTgyMjktNDU5YS1iZmFlLTdmN2I5Nzc4NWM0NWo%3D&page[limit]=10",
    "prev": null
  },
```

Reply:
That is so strange... https://hexdocs.pm/plug/Plug.Conn.html#request_url/1

We get `request.url` from `Plug.Conn.request_url/1`, which should be the full request url. Can you confirm that `Plug.Conn.request_url/1` is returning the incorrect value there?

Reply:
Sorry I did not noticedthat the fix https://github.com/ash-project/ash_json_api/commit/c9993777b4bc71f250fb542b2ba18961a397554e was not released yet.
So the it's normal I got the bare url because `endpoint.url()` is called.
Pinning the deps to main solve all the issues I encountered with links.